cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -103,6 +103,22 @@ func RemoveStackedEtcdMemberFromCluster(client clientset.Interface, cfg *kubeadm
 		return err
 	}
 
+	members, err := etcdClient.ListMembers()
+	if err != nil {
+		return err
+	}
+	// If this is the only remaining stacked etcd member in the cluster, calling RemoveMember()
+	// is not needed.
+	if len(members) == 1 {
+		etcdClientAddress := etcdutil.GetClientURL(&cfg.LocalAPIEndpoint)
+		for _, endpoint := range etcdClient.Endpoints {
```

Comment:
Should we check etcdClientAddress == member.clientURLs instead of relying on endpoints?

Reply:
Thanks @fabriziopandini for the review.
`etcdutil.NewFromCluster` has read all etcd endpoints from etcd and the `etcdClient.Endpoints` are [got from ClientURLs of all members](https://github.com/etcd-io/etcd/blob/747ff75c96df87530bcd8b6b02d1160c5500bf4e/clientv3/client.go#L170-L181).
https://github.com/kubernetes/kubernetes/blob/2f38e1b130ea75704bf17316a9c5e89bb36e2d54/cmd/kubeadm/app/util/etcd/etcd.go#L113-L118

And currently the Member struct returned here has only a few information and doesn't have clientURLs: https://github.com/kubernetes/kubernetes/blob/2f38e1b130ea75704bf17316a9c5e89bb36e2d54/cmd/kubeadm/app/util/etcd/etcd.go#L254-L259
So checking `etcdClientAddress` in `etcdClient.Endpoints` should have the same result as your suggestion and doesn't require struct change?

Reply:
ok, given that the goal of this check is to check this is actually the last member in the cluster.